src/couch_replicator_scheduler.erl

```diff
@@ -259,25 +267,57 @@ remove_job_int(#job{} = Job) ->
     ets:delete(?MODULE, Job#job.id).
 
 
+-spec job_count() -> non_neg_integer().
+job_count() ->
+    ets:info(?MODULE, size).
+
+
 -spec running_job_count() -> non_neg_integer().
 running_job_count() ->
-    ets:info(?MODULE, size) - pending_job_count().
+    MatchSpec = [{#job{pid = '$1', _='_'},
+        [{is_pid,'$1'}],
+        [true]}
+    ],
+    ets:select_count(?MODULE, MatchSpec).
 
 
 -spec running_jobs() -> [#job{}].
 running_jobs() ->
-    ets:tab2list(?MODULE) -- pending_jobs().
+    MatchSpec = [{#job{pid = '$1', _='_'},
+        [{is_pid,'$1'}],
+        ['$_']}
+    ],
+    ets:select(?MODULE, MatchSpec).
 
 
 -spec pending_job_count() -> non_neg_integer().
 pending_job_count() ->
-    MatchSpec = [{#job{pid='$1', _='_'}, [{'not', {'is_pid', '$1'}}], [true]}],
+    Now = unix_now(),
+    % Select jobs where the pid is undefined and the exponential backoff
+    % threshold has been crossed.
+    MatchSpec = [{#job{pid = '$1',errored_at = '$2',errors = '$3', _='_'},
+        [{'not',{is_pid,'$1'}},
```

Comment:
Maybe in the head just say `#job{pid = undefined}` to save an extra guard. Should probably also update typespec of job records' pid field to be `pid() | undefined`